src/compiler/moduleSpecifiers.ts

```diff
@@ -105,6 +105,14 @@ namespace ts.moduleSpecifiers {
         const moduleSourceFile = getSourceFileOfNode(moduleSymbol.valueDeclaration || getNonAugmentationDeclaration(moduleSymbol));
         const modulePaths = getAllModulePaths(importingSourceFile.path, moduleSourceFile.originalFileName, host);
 
+        const existingSpecifier = forEach(modulePaths, modulePath => forEach(

```

Comment:
importFixes.ts calls this function, and will never find this `existingSpecifier` since it takes a different approach altogether if there was an existing specifier the auto import could reuse. It’s probably not very expensive, but it would be nice if we could skip this in that case.

Reply:
Looked into this.. I don't think this is expensive since its not going to find the import in the file include reason. the importing file is already in program so thats also not good quick check. Adding another flag or optionality adds to maintenance cost later so would like to keep this.